application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -128,18 +130,24 @@ public BotCore(@NotNull JDA jda, @NotNull Database database, @NotNull Config con
                 })));
 
         if (logger.isInfoEnabled()) {
-            logger.info("Available commands: {}", nameToSlashCommands.keySet());
+            logger.info("Available user interactors: {}", nameToInteractor.keySet());
         }
     }
 
     @Override
     public @NotNull Collection<SlashCommand> getSlashCommands() {
-        return Collections.unmodifiableCollection(nameToSlashCommands.values());
+        return nameToInteractor.values()
+            .stream()
+            .filter(SlashCommand.class::isInstance)
+            .map(SlashCommand.class::cast)
+            .toList();
     }
 
     @Override
     public @NotNull Optional<SlashCommand> getSlashCommand(@NotNull String name) {
-        return Optional.ofNullable(nameToSlashCommands.get(name));
+        return Optional.ofNullable(nameToInteractor.get(name))
+            .filter(SlashCommand.class::isInstance)
+            .map(SlashCommand.class::cast);
```

Comment:
I honestly want a second map for SlashCommands, this is really under performant.

Reply:
Why? Whats the issue that you see? I dont like maintaining a map and a second map that is just a subview of it.

Reply:
The first map you loop through all items, filter and map.
The second, you still filter and map.

I personally find that a waste of performance, there's nothing to "maintain"
The map gets created on start-up, after that it doesn't anymore.

But now I just see less performance, and more code, and in return?

Reply:
The first can be solved by creating a restricted view of the set (that just blends out the stuff with the wrong type), if you really care that much. The map is small though and the method is only used once by reload - so its not really a big deal anyways.

The second one is not an iteration at all, the filter and map call are on Optional, so the second one is perfectly fine.